src/address.rs

```diff
@@ -42,9 +46,30 @@ impl<A: Actor, M: Message> Default for SendFutureInner<A, M> {
     }
 }
 
-pub(crate) fn poll_rx<T>(rx: &mut Receiver<T>, ctx: &mut Context) -> Poll<Result<T, Error>> {
-    rx.poll_unpin(ctx)
-        .map(|r| r.map_err(|_| Error::Disconnected))
+pub(crate) fn poll_rx<T>(
+    rx: &mut Receiver<T>,
+    #[cfg(feature = "timeout")] rx_timed_out: &mut Receiver<()>,
+    ctx: &mut Context,
+) -> Poll<Result<T, Error>> {
+    #[cfg(not(feature = "timeout"))]
+    let res = rx
+        .poll_unpin(ctx)
+        .map(|r| r.map_err(|_| Error::Disconnected));
+
+    #[cfg(feature = "timeout")]
+    let res = {
+        let result_res = rx.poll_unpin(ctx);
+        let timed_out_res = rx_timed_out.poll_unpin(ctx);
+
+        match (result_res, timed_out_res) {
+            (Poll::Ready(Ok(result)), _) => Poll::Ready(Ok(result)),
```

Comment:
Would it be worth it to write out the seconds futures state instead of `_` here?

Reply:
It is intentional that the first match arm corresponds to the result future being ready and whatever the result is of the timeout future. I want to always succeed if we have the result of the first channel, even if the second future failed for whatever reason.

For instance, if the second future reports that the timeout has passed but the result is actually ready, we obviously should ignore the timeout and return the result. The timeout is only there as a mechanism to stop handlers that have been running for too long _without success_.

In practice it may be impossible to ever have a situation where both futures are `Ready(Ok(_))` since we're using `future::select` in `envelope.rs`. What certainly does happen (consistently) is that the `timed_out_channel` is dropped as soon as the `result_channel` returns a value. The `Drop` implementation of the `catty` channel construct actually causes the channel to produce an `Err(Disconnected)`. So when we match against the `catty::Receivers` and we have a successful result for the `result_channel`, we will have an error for the second one. Which is why we must ignore it for this to work anyway.